src/io.nim

```diff
@@ -0,0 +1,248 @@
+# Stint
+# Copyright 2018 Status Research & Development GmbH
+# Licensed under either of
+#
+#  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE) or http://www.apache.org/licenses/LICENSE-2.0)
+#  * MIT license ([LICENSE-MIT](LICENSE-MIT) or http://opensource.org/licenses/MIT)
+#
+# at your option. This file may not be copied, modified, or distributed except according to those terms.
+
+import
+  ./private/datatypes,
+  ./private/int_negabs,
+  ./private/initialization,
+  ./private/[as_words, as_signed_words],
+  ./int_public, ./uint_public,
+  typetraits, algorithm
+
+func stuint*[T: SomeInteger](n: T, bits: static[int]): StUint[bits] {.inline.}=
+  assert n >= 0.T
+  when result.data is UintImpl:
+    when getSize(n) > bits:
+      # To avoid a costly runtime check, we refuse storing into StUint types smaller
+      # than the input type.
+      raise newException(ValueError, "Input " & $n & " (" & $T &
```

Comment:
Hrmm.. Why not just fail the compilation altogether?

Reply:
They were supposed to be, see comment "Use a smaller input type instead. This is a compile-time check" ;). But I forgot to change from exception to static asserts. Fixed.

Reply:
Actually I get, "cannot evaluate at compile-time n" so I have to resort to something else